src/components/Time.jsx

```diff
@@ -7,17 +7,39 @@ import Tooltip from './Tooltip'
 
 import TimeUtils from './utils/TimeUtils'
 
+const DELAY = 60000
+
 class Time extends React.Component {
     constructor(props, context) {
         super(props, context)
 
+        this.refresh_timeout = null
+
         this.state = {
             hovered: false,
             mouseX: null,
             mouseY: null
         }
     }
 
+    componentDidMount() {
+        if (this.props.refresh) {
+            this._refresh()
+        }
+    }
+
+    componentDidUpdate() {
+        clearTimeout(this.refresh_timeout)
+
+        if (this.props.refresh) {
+            this._refresh()
+        }
+    }
```

Comment:
We only want to do this if the `refresh` prop changed from `false` to `true`, and conversely to do the opposite if it changed from `true` to `false`:

```js
componentDidUpdate(prevProps) {
  if (!prevProps.refresh && this.props.refresh) {
    this._refresh()
  }
  if (prevProps.refresh && !this.props.refresh) {
    clearTimeout(this.refresh_timeout)
  }
}
```